lib/java-shade.gradle

```diff
@@ -358,16 +368,23 @@ private Configuration configureShadedTestImplementConfiguration(
                 }
                 // Do not use `project.dependencies.add(name, dep)` that discards the classifier of
                 // a dependency. See https://github.com/gradle/gradle/issues/23096
-                project.configurations.getByName(shadedJarTestImplementation.name).dependencies.add(dep)
+                shadedJarTestImplementation.dependencies.add(dep)
             }
         }
     }
 
     // Recurse into the project dependencies.
     projectDependencies.each { ProjectDependency dep ->
+        if (!dep.dependencyProject.hasFlag('relocate')) {
+            shadedJarTestImplementation.dependencies.add(
+                    project.dependencies.project(path: dep.dependencyProject.path))
```

Comment:
Question) I haven't been able to look into this in detail. If the `project.dependencies` contains a dependency that should've been shaded, will this case be covered correctly?

Reply:
I believe so. An exception will be raised from here:
https://github.com/line/gradle-scripts/pull/181/files#diff-5c14d66c4edef5e74eac3b83b26ba720aafc5b5b008248dea72d951aa64170deR357-R360

Reply:
Checked that transient dependencies aren't taken care of for some reason previously as well which made me confused.

I guess functionally there is no difference then.